### PR TITLE
[SYCL] Drop get_native class functions

### PR DESCRIPTION
**Pull request by sergey-semenov**
Drop deprecated get_native class functions in favor of get_native
free functions.

### CODE REVIEW

**Comment by sergey-semenov** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1113

File: sycl/include/sycl/backend.hpp [hunk 1]
```diff
@@ -127,7 +127,19 @@ auto get_native(const SyclObjectT &Obj)
...
                               PI_ERROR_INVALID_OPERATION);
   }
-  return Obj.template get_native<BackendName>();
+  return reinterpret_cast<backend_return_t<BackendName, SyclObjectT>>(
+      Obj.getNative());
+}
+
+template <backend BackendName, bundle_state State>
+auto get_native(const kernel_bundle<State> &Obj)
+    -> backend_return_t<BackendName, kernel_bundle<State>> {
```
Comment (aelovikov-intel) [inline sycl/include/sycl/backend.hpp:136]: Do we have style guide for this spelling of the return type? I don't think I've seen it anywhere in the code before...

Comment (sergey-semenov) [reply sycl/include/sycl/backend.hpp:136]: I don't think we do, and, personally, I don't see the point of it here and would rather get rid of it, just did it for consistency with the existing code in this file. I don't mind making the change as part of this PR if you agree.

Comment (aelovikov-intel) [reply sycl/include/sycl/backend.hpp:136]: > for consistency with the existing code in this file

Missed that, thanks for the explanation.

> I don't mind making the change as part of this PR if you agree.

I think that should be a standalone PR having no ties to this one (i.e. shouldn't be even required for you).

**Comment by againull** [comment]
Sorry for delayed review. Changes look good to me, but could you please fix failing test:
Failed Tests (1):
  SYCL :: basic_tests/interop-cuda-experimental.cpp

**Comment by sergey-semenov** [comment]
@againull Done, please take a look.

**Review by againull — APPROVED** [review]

**Comment by sergey-semenov** [comment]
@againull Sorry, I accidentally left one change uncommitted, so the test still fails, I'll open a new PR ASAP.

**Comment by sergey-semenov** [comment]
@againull See https://github.com/intel/llvm/pull/6504

**Comment by aelovikov-intel** [comment]
I also have a failure on "SYCL :: DeprecatedFeatures/cuda_program_interop.cpp" like this (https://github.com/intel/llvm/runs/7617823189?check_suite_focus=true):

```
/__w/llvm/llvm/llvm_test_suite/SYCL/DeprecatedFeatures/cuda_program_interop.cpp:15:29: error: no member named 'get_native' in 'sycl::program'
  auto NativeProgram = Prog.get_native<sycl::backend::ext_oneapi_cuda>();
                       ~~~~ ^
```

@sergey-semenov , can you confirm this was caused by your change?

**Comment by againull** [comment]
> I also have a failure on "SYCL :: DeprecatedFeatures/cuda_program_interop.cpp" like this (https://github.com/intel/llvm/runs/7617823189?check_suite_focus=true):
> 
> ```
> /__w/llvm/llvm/llvm_test_suite/SYCL/DeprecatedFeatures/cuda_program_interop.cpp:15:29: error: no member named 'get_native' in 'sycl::program'
>   auto NativeProgram = Prog.get_native<sycl::backend::ext_oneapi_cuda>();
>                        ~~~~ ^
> ```
> 
> @sergey-semenov , can you confirm this was caused by your change?

I think, Sergey fixed this in https://github.com/intel/llvm-test-suite/pull/1125

**Pull request merged**